<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>278</width>
    <height>601</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>FreeMono</family>
    <pointsize>12</pointsize>
    <weight>75</weight>
    <bold>true</bold>
   </font>
  </property>
  <property name="windowTitle">
   <string>Generator</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="pb_loadImage">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>10</y>
      <width>171</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>FreeMono</family>
      <pointsize>12</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Load Image</string>
    </property>
   </widget>
   <widget class="QStackedWidget" name="stackedWidget_values">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>210</y>
      <width>261</width>
      <height>341</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Box</enum>
    </property>
    <property name="currentIndex">
     <number>4</number>
    </property>
    <widget class="QWidget" name="Gaussian">
     <widget class="QSpinBox" name="spinBox_k1">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_k2">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>60</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_kernelG">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Kernel Size</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_gb">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>171</width>
        <height>17</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="text">
       <string>Gaussian Blur</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_sigx">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_sigy">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>210</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_sigmaX">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>110</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="text">
       <string notr="true">Sigma X</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_sigmaY">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>190</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Sigma Y</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="Blur">
     <widget class="QLabel" name="label_kernelB">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Kernel Size</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_Bk2">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>60</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_blur">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>171</width>
        <height>17</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="text">
       <string>Blur</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_Bk1">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="MedianBlur">
     <widget class="QLabel" name="label_medianBlur">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>171</width>
        <height>17</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="text">
       <string>Median Blur</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_ksizeMB">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>61</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_ksizeMB">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="text">
       <string notr="true">kSIZE</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="BilateralFilter">
     <widget class="QLabel" name="label_bilFilter">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>171</width>
        <height>17</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="text">
       <string>Bilateral Filter</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_BFd">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>71</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>255</number>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_BFsigmaColor">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>101</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <double>1.000000000000000</double>
      </property>
      <property name="maximum">
       <double>255.000000000000000</double>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_BFsigmaSpace">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>180</y>
        <width>101</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <double>1.000000000000000</double>
      </property>
      <property name="maximum">
       <double>255.000000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_BFd">
      <property name="geometry">
       <rect>
        <x>16</x>
        <y>40</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>d</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_BFSC">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Sigma Color</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_BFSS">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>160</y>
        <width>121</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Sigma Space</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page">
     <widget class="QLabel" name="label_convolution">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>171</width>
        <height>17</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="text">
       <string>Apply Convolution</string>
      </property>
     </widget>
     <widget class="QFrame" name="frame">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>221</width>
        <height>161</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QSpinBox" name="spinBox_k00">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k01">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>20</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k02">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>20</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k03">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>20</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k04">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>20</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k10">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>43</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k13">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>43</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k12">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>43</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k14">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>43</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k11">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>43</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k20">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>66</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k23">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>66</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k22">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>66</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="value">
        <number>1</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k24">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>66</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k21">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>66</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k30">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>89</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k33">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>89</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k32">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>89</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k34">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>89</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k31">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>89</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k40">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>112</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k43">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>112</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k42">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>112</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k44">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>112</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="spinBox_k41">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>112</y>
         <width>42</width>
         <height>26</height>
        </rect>
       </property>
       <property name="minimum">
        <number>-255</number>
       </property>
       <property name="maximum">
        <number>255</number>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="pb_saveImageConv">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>240</y>
        <width>181</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Save Image</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pb_saveConfigConv">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>280</y>
        <width>181</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Save Config Values</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page_2"/>
    <widget class="QWidget" name="page_3"/>
    <widget class="QWidget" name="page_4"/>
    <widget class="QWidget" name="page_5"/>
    <widget class="QWidget" name="page_6"/>
   </widget>
   <widget class="QPushButton" name="pb_viewOriginalImage">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>50</y>
      <width>171</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>View Image</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_functions">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>140</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>-1</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_selectFunction">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>120</y>
      <width>151</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Select Function</string>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>278</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>Menu</string>
    </property>
    <addaction name="separator"/>
    <addaction name="separator"/>
    <addaction name="loadValues"/>
    <addaction name="saveAllConfigs"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="loadValues">
   <property name="text">
    <string>Load File Values</string>
   </property>
  </action>
  <action name="saveAllConfigs">
   <property name="text">
    <string>Save all Configs</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
